### PR TITLE
Implementation of the debug_getRawReceipts method

### PR DESCRIPTION
**Pull request by novosandara**
# Description

Implementation of the debug_getRawReceipts method

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

# Documentation update

# Additional comments

### CODE REVIEW

File: jsonrpc/debug_endpoint.go [hunk 1]
```diff
@@ -295,6 +298,33 @@ func (d *Debug) GetRawTransaction(txHash types.Hash) (interface{}, error) {
...
+			receipts, err := d.store.GetReceiptsByHash(header.Hash)
+			if err != nil {
+				return nil, err
+			}
+
+			result := make([][]byte, len(receipts))
+
+			for i, receipt := range receipts {
+				b := receipt.MarshalRLP()
+				result[i] = b
```
Comment (Stefan-Ethernal) [inline jsonrpc/debug_endpoint.go]: Minor optimization, if you find it necessary (code is still readable enough IMO):

```suggestion
				result[i] = receipt.MarshalRLP()
```

File: jsonrpc/debug_endpoint_test.go [hunk 2]
```diff
@@ -1070,6 +1075,97 @@ func TestGetRawTransaction(t *testing.T) {
...
+			t.Parallel()
+
+			endpoint := NewDebug(test.store, 100000)
+
+			res, err := endpoint.GetRawReceipts(test.filter)
+
+			assert.Equal(t, test.result, res)
+
+			if test.err {
+				assert.Error(t, err)
```
Comment (Stefan-Ethernal) [inline jsonrpc/debug_endpoint_test.go]: Consider asserting for exact error or that the error string contains a certain substring. WDYT?

File: jsonrpc/debug_endpoint_test.go [hunk 3]
```diff
@@ -1070,6 +1075,97 @@ func TestGetRawTransaction(t *testing.T) {
...
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			endpoint := NewDebug(test.store, 100000)
+
+			res, err := endpoint.GetRawReceipts(test.filter)
+
+			assert.Equal(t, test.result, res)
```
Comment (Stefan-Ethernal) [inline jsonrpc/debug_endpoint_test.go]: Try and see whether linter will complain, but not necessary to add empty line after each line IMO:

```suggestion
			endpoint := NewDebug(test.store, 100000)
			res, err := endpoint.GetRawReceipts(test.filter)
			assert.Equal(t, test.result, res)
```

File: jsonrpc/debug_endpoint_test.go [hunk 4]
```diff
@@ -1070,6 +1075,97 @@ func TestGetRawTransaction(t *testing.T) {
...
+
+			if test.err {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
```
Comment (Stefan-Ethernal) [inline jsonrpc/debug_endpoint_test.go]: Use `require`, instead of `assert` in the entire unit test. WDYT?

**Review by Stefan-Ethernal — APPROVED** [review]
Generally LGTM!

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**